gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -1086,12 +1086,15 @@ _gtfc_setprop_INPUT (GstTensorFilterPrivate * priv,
     num_dims = gst_tensors_info_parse_dimensions_string (&prop->input_meta,
         g_value_get_string (value));
 
-    if (prop->input_meta.num_tensors > 0 &&
-        prop->input_meta.num_tensors != num_dims) {
-      ml_logw ("Invalid input-dim, given param does not match with old value.");
-    }
+    if (num_dims > 0) {
+      if (prop->input_meta.num_tensors > 0 &&
+          prop->input_meta.num_tensors != num_dims) {
+        ml_logw
+            ("Invalid input-dim, given param does not match with old value.");
+      }
 
-    prop->input_meta.num_tensors = num_dims;
+      prop->input_meta.num_tensors = num_dims;
```

Comment:
Is it ok to update ```prop->input_meta.num_tensors``` when ```num_dims == 0```? (and same for other similar variables)

Reply:
> Is it ok to update prop->input_meta.num_tensors when num_dims == 0? (and same for other similar variables)

That is exactly what this patch prevents. If `num_dims` interpreted with the given property is 0, then it does not update the `prop->input_meta.num_tensors`.

However, if multiple properties have different dimensions, then the property setting the last dimension wins with the current implementation (supporting changing the num_tensors of an already configured tensor_filter).